src/main/scala/chisel3/util/SparseVec.scala

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util
+
+import chisel3._
+import chisel3.experimental.SourceInfo
+import chisel3.util.{log2Up, BitPat, Mux1H, PriorityMux}
+import chisel3.util.experimental.decode.{decoder, TruthTable}
+import chisel3.util.experimental.InlineInstance
+import scala.collection.SeqMap
+import scala.collection.mutable.{ArrayBuffer, LinkedHashMap, LinkedHashSet}
+
+private[this] object Detail {
```

Comment:
What does `private[this]` do when it's not inside of a class? Is it private to package `util` or private to this file?

Reply:
I thought this was file private, but it's apparently object private. This makes it accessible inside `chisel3.util` which was not the intention. I'll move this around.